src/components/rich-radio/RichRadio.tsx

```diff
@@ -118,6 +118,10 @@ export const RichRadio = <T extends string>({
 }: RichRadioProps<T>) => {
   const [selectedValue, setSelectedValue] = useState(value);
 
+  useEffect(() => {
+    setSelectedValue(value);
+  }, [value]);
+
```

Comment:
It's interesting 🤔 Do you mean to change/adjust value in RichRadio outside of the Rich radio component after the Rich radio is rendered?

Reply:
In the event of a parent fetching data, the initial value will be incorrect. That means on mount the rich radio has the incorrect value. When the value comes in, it needs to reset the state so that it will show the selected option as active.

Reply:
Gotcha, thanks for clarifying that 😉